skcosmo/preprocessing/flexible_scaler.py

```diff
@@ -25,25 +25,33 @@ def __init__(self, with_mean=True, with_std=True, column_wise=False, tol=1e-15):
         self.n_samples_seen_ = 0
         self.tol = tol
 
-    def fit(self, X, y=None):
+    def fit(self, X, y=None, sample_weight=None):
         """Compute mean and scaling to be applied for subsequent normalization.
 
         :param X: Matrix
         :type X: ndarray
         :param y: ignored
+        :param sample_weight: weights for each sample. Sample weighting can be used to center (and scale) data using a weighted mean. Weights are normalized before preprocessing.
+        :type sample_weight: array of shape (n_samples,)
 
         :return: itself
         """
 
         self.n_samples_seen_, self.n_features_ = X.shape
+
+        if sample_weight is not None:
+            sample_weight = _check_sample_weight(sample_weight, X, dtype=X.dtype)
+            sample_weight = sample_weight / np.sum(sample_weight)
```

Comment:
Docstring is confusing -- it can read that weights _should_ be normalized before passing. What happens if someone wants them non-normalized? Should we allow that?

Reply:
I have tweaked the docstrings slightly to clarify that the weights are normalized automatically. I think the weights should definitely be normalized. For instance, if you didn't normalize the weights and assigned each sample a weight of 2, you would end up centering by 2x the mean.